### PR TITLE
test(core): cover detectTests.ts qualify/validate/parse pipeline (Phase 6)

### PR DESCRIPTION
**Pull request by hawkeyexl**
## Phase 6 of: root coverage ratchet

Adds `test/detecttests-coverage.test.js` (31 cases) covering `src/core/detectTests.ts` — the runner's test-detection pipeline (qualify input files → validate sources → parse into specs).

### Approach
Drives the **real** pipeline through the exported `detectTests` / `parseTests` / `generateSpecId` with hermetic **tmpdir fixtures** and a config built via `setConfig` (modeled on the existing `spec-location.test.js` / `run-artifacts.test.js`). No source changes.

### Covered
- **qualifyFiles** — no-input short-circuit, directory recursion, `node_modules` skip, `recursive:false`, dedupe, missing-input `statSync` catch, single file, `_phase` stamping.
- **isValidSourceFile** — non-spec/non-object JSON + non-spec YAML skips; missing `before`/`after` under `relativePathBase` `file` and `cwd`; disallowed extension; before/after merge.
- **parseTests** — markdown inline steps, empty-spec drop, valid JSON/YAML specs, malformed-JSON fallback, `runShell` success + invalid-step skip, testId collision suffix, `_phase` default.
- **generateSpecId** — inside/outside-cwd, sibling-prefix edge, sanitization, determinism.

Isolated coverage: **~75% → 83.94% lines / 86.6% branches**.

### Left for E2E (noted in the test file)
- `processDitaMap` — shells out to the external **`dita` CLI**; not hermetically testable.
- Heretto integration + URL input — network/API.
- Defensive post-merge re-validation branches — unreachable through the real pipeline (upstream `readFile` already validates).

### Notes
- Hermetic: tmpdir fixtures only (cleaned per test), no network/spawn. `this.timeout(30000)` covers the first-call AJV `config_v3` compilation (per the Phase 4 timeout lesson). Tests pass deterministically across repeated runs.
- The `coverage-thresholds.json` bump lands once the `Coverage ratchet (root)` job measures the new full-suite number.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added a new Mocha test suite covering test detection and parsing across Markdown/text, JSON, and YAML.
  * Validates file qualification behavior (recursive/non-recursive, deduping, missing or glob-like inputs) and applies before/main/after phase stamping.
  * Verifies source validation (extension filtering, missing before/after handling, and merging referenced steps).
  * Exercises parsing edge cases, including `runShell` handling, malformed templates, and deterministic spec/test ID generation.

* **Chores / CI**
  * Increased coverage baseline thresholds for lines, statements, and branches.

### CODE REVIEW

File: test/detecttests-coverage.test.js [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+      // a real path, so statSync throws and the source is skipped (catch branch).
+      write("one.json", specJSON("https://a.com"));
+      const real = write("two.json", specJSON("https://b.com"));
+      const config = await makeConfig({
+        input: [path.join(tmpDir, "*.json").split(path.sep).join("/"), real],
+      });
+      const specs = await detectTests({ config });
+      // Only the real explicit file resolves; the glob string is skipped.
+      assert.equal(specs.length, 1);
+    });
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/detecttests-coverage.test.js:115]: Keeping both, intentionally: they share the statSync catch branch but assert different contracts — one documents that a missing file path is skipped, the other documents that qualifyFiles does NOT expand glob patterns (a non-obvious behavior). The runtime cost is negligible (no I/O beyond a failed stat), and the distinct intent aids the next reader.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
